File: programs/mandel/Mandel.java
public class Mandel {
    public static void main(String[] args) {
        int n = 400;
        int limit = 1000;
        long total = 0;
        for (int y = 0; y < n; y++) {
            double ci = -1.5 + (double) y * (3.0 / (double) n);
            for (int x = 0; x < n; x++) {
                double cr = -2.0 + (double) x * (3.0 / (double) n);
                double zr = 0.0, zi = 0.0;
                int iter = 0;
                while (zr * zr + zi * zi <= 4.0 && iter < limit) {
                    double tr = zr * zr - zi * zi + cr;
                    zi = 2.0 * zr * zi + ci;
                    zr = tr;
                    iter++;
                }
                total += iter;
            }
        }
        System.out.println("mandel " + n + " total = " + total);
    }
}

File: programs/mandel/mandel.go
package main

import "fmt"

func main() {
    n := 400
    limit := 1000
    var total int64
    for y := 0; y < n; y++ {
        ci := -1.5 + float64(y)*(3.0/float64(n))
        for x := 0; x < n; x++ {
            cr := -2.0 + float64(x)*(3.0/float64(n))
            zr, zi := 0.0, 0.0
            iter := 0
            for zr*zr+zi*zi <= 4.0 && iter < limit {
                tr := zr*zr - zi*zi + cr
                zi = 2.0*zr*zi + ci
                zr = tr
                iter++
            }
            total += int64(iter)
        }
    }
    fmt.Printf("mandel %d total = %d\n", n, total)
}
